sdk/src/lib.rs

```diff
@@ -239,21 +249,33 @@ impl TestRustWorker {
             Result::<_, anyhow::Error>::Ok(self)
         };
 
-        let mut myself = pollers.await?;
+        let myself = pollers.await?;
         while let Some(h) = myself.workflow_half.join_handles.next().await {
             h??;
         }
         before_shutdown();
         myself.core.shutdown().await;
+        myself.workflow_half.workflows.clear();
         Ok(())
     }
 
     /// Drives all workflows & activities until they have all finished, repeatedly polls server to
     /// fetch work for them.
-    pub async fn run_until_done(self) -> Result<(), anyhow::Error> {
+    pub async fn run_until_done(&mut self) -> Result<(), anyhow::Error> {
         self.run_until_done_shutdown_hook(|| {}).await
     }
 
+    /// Temporarily swap the core implementation used by this worker. When the returned value is
+    /// dropped, the old core implementation will be restored. This can be used to run against
+    /// mocked-out core instances.
+    pub fn swap_core(&mut self, other_core: Arc<dyn Core>) -> impl DerefMut<Target = Self> + '_ {
```

Comment:
Why would we want to swap the core if we have `register_replay_worker`?

Reply:
Because there is a shared client Core uses for some operations that isn't tied to a worker, that client needs to be changed out so it's a mock one and not a real one.

Reply:
I saw where it's used, I guess my question is why not make a new core with the new worker?

Reply:
Could do. This just ends up being easier for tests where you want to do it "real" and then download the replay. In either case this only gets used internally in integ tests. Now that the test_utils crate isn't needed for replay no one else sees this.